auth/overview/oauth_client.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package overview contains Google Cloud authentication overview snippets.
+// https://cloud.google.com/docs/authentication/
+package overview
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/option"
+	"google.golang.org/api/pubsub/v1"
+)
+
+// [START auth_overview_oauth_client]
+
+// oauthClient shows how to OAuth client ID to authenticate as an end-user.
+func oauthClient() {
+	ctx := context.Background()
+
+	// Please make sure the redirect URL is the same as the one you specified when you
+	// created the client ID.
+	redirectURL := os.Getenv("OAUTH2_CALLBACK")
+	if redirectURL == "" {
+		redirectURL = "your redirect url"
+	}
+	config := &oauth2.Config{
+		ClientID:     "your-client-id",
+		ClientSecret: "your-client-secret",
+		RedirectURL:  redirectURL,
+		Scopes:       []string{"email", "profile"},
+		Endpoint:     google.Endpoint,
+	}
+
+	// Dummy authorization flow to read auth code from stdin
+	authURL := config.AuthCodeURL("your state")
+	fmt.Printf("Follow the link in your browser to obtain auth code: %s", authURL)
+
+	// Read the authentication code from the command line
+	var code string
+	fmt.Scanln(&code)
```

Comment:
Clarifying the comment about making this an argument, does this need to be read from standard in? Knowing very little about OAuth, I don't know what `code` is, either. Could we do:

```
func oauthClient(code string) error {
```

(This also makes it return an error, which you'll need to avoid `log.Fatal`.)

Reply:
The authentication code is only used to exchange for an access token, I don't think it should be exposed as an argument in this case.